### ISSUE TITLE
3proxy-0.8.13 doesn't forward IPv6 addresses in udppm

### ISSUE THREAD

**Issue by phantomcraft** [z3APA3A/3proxy#530]
Easy to reproduce:

```
user@localhost:/dev/shm/3proxy-0.8.13$ ./src/3proxy << EOT
> log
> auth iponly
> allow *
> udppm -i127.0.0.1 1053 [::1] 53
> EOT
201016160855.387 1053 00000 - 127.0.0.1:1053 0.0.0.0:0 0 0 0 Accepting connections [87572/1910626048]
201016160922.160 1053 00097 - 127.0.0.1:50717 ::1:53 0 0 0 
201016160927.162 1053 00097 - 127.0.0.1:50717 ::1:53 0 0 0 
201016160932.162 1053 00097 - 127.0.0.1:50717 ::1:53 0 0 0 
```

```
user@localhost:~$ dig @127.0.0.1 -p 1053 t.co
; <<>> DiG 9.16.6-Debian <<>> @127.0.0.1 -p 1053 t.co
; (1 server found)
;; global options: +cmd
;; connection timed out; no servers could be reached
```
===================================================

```
user@localhost:/dev/shm/3proxy-0.8.13$ ./src/udppm -l -i127.0.0.1 1053 [::1] 53
201016161143.820 1053 00000 - 127.0.0.1:1053 0.0.0.0:0 0 0 0 Accepting connections [87668/751523648]
201016161146.516 1053 00097 - 127.0.0.1:40475 ::1:53 0 0 0 
201016161151.518 1053 00097 - 127.0.0.1:40475 ::1:53 0 0 0 
201016161156.518 1053 00097 - 127.0.0.1:40475 ::1:53 0 0 0 
```

```
user@localhost:~$ dig @127.0.0.1 -p 1053 t.co
; <<>> DiG 9.16.6-Debian <<>> @127.0.0.1 -p 1053 t.co
; (1 server found)
;; global options: +cmd
;; connection timed out; no servers could be reached
```

**Comment by phantomcraft** [comment]
In 3proxy-0.9.0-rc binary, an endless loop happens:

```
user@localhost:/dev/shm/3proxy-0.9.0-rc$ ./bin/3proxy << EOT
log
auth iponly
allow *
udppm -i127.0.0.1 1053 [::1] 53
EOT
201016162020.440 1053 00000 - 127.0.0.1:1053 0.0.0.0:0 0 0 0 Accepting connections [126279/1142617856]
201016162024.287 1053 00100 - 127.0.0.1:1053 0.0.0.0:0 0 0 0 
201016162024.288 1053 00100 - 0.0.0.0:0 0.0.0.0:0 0 0 0 
201016162024.290 1053 00100 - 0.0.0.0:0 0.0.0.0:0 0 0 0 
[...]
201016162024.291 1053 00100 - 0.0.0.0:0 0.0.0.0:0 0 0 0 
201016162024.293 1053 00100 - 0.0.0.0:0 0.0.0.0:0 0 0 0 
```

===============================================

The same with udppm binary (and the proccess can't be killed by CTRL+C):

```
user@localhost:/dev/shm/3proxy-0.9.0-rc$ ./bin/udppm -l -i127.0.0.1 1053 [::1] 53
201016162237.238 1053 00000 - 127.0.0.1:1053 0.0.0.0:0 0 0 0 Accepting connections [152148/4191541056]
201016162240.581 1053 00100 - 127.0.0.1:1053 0.0.0.0:0 0 0 0 
201016162240.587 1053 00100 - 0.0.0.0:0 0.0.0.0:0 0 0 0 
201016162240.589 1053 00100 - 0.0.0.0:0 0.0.0.0:0 0 0 0 
201016162240.590 1053 00100 - 0.0.0.0:0 0.0.0.0:0 0 0 0 
[...]
201016162300.323 1053 00100 - 0.0.0.0:0 0.0.0.0:0 0 0 0 
201016162300.324 1053 00100 - 0.0.0.0:0 0.0.0.0:0 0 0 0 
201016162300.325 1053 00100 - 0.0.0.0:0 0.0.0.0:0 0 0 0 
201016162300.327 1053 00100 - 0.0.0.0:0 0.0.0.0:0 0 0 0 
Killed
```

**Comment by z3APA3A** [comment]
Yes, IPv6 support in 0.8 was incomplete, there is no longer development in 0.8, 0.9 will be released next week.

**Comment by z3APA3A** [comment]
P.S. I've added few patches to 0.9 branch and devel (10) branch
624c3e6ae1209df718818742807459c814992e5b
01ec2f12d59c9445b6a02ed32c486615886003e9

but currently command line utilities use -4 by default, add -46 flag to support IPv6.

**Comment by phantomcraft** [comment]
Even with -46 the problem persists:

```
user@localhost:/dev/shm/3proxy-0.8.13$ ./src/udppm -l -46 -i127.0.0.1 1053 [::1] 1053
201016233400.992 1053 00000 - 127.0.0.1:1053 0.0.0.0:0 0 0 0 Accepting connections [175313/3306362688]
201016233411.596 1053 00097 - 127.0.0.1:56232 ::1:1053 0 0 0 
201016233416.596 1053 00097 - 127.0.0.1:56232 ::1:1053 0 0 0 
201016233421.596 1053 00097 - 127.0.0.1:56232 ::1:1053 0 0 0 
```
```

user@localhost:/dev/shm/3proxy-0.8.13$ dig @127.0.0.1 -p 1053 t.co
; <<>> DiG 9.16.6-Debian <<>> @127.0.0.1 -p 1053 t.co
; (1 server found)
;; global options: +cmd
;; connection timed out; no servers could be reached
```

**Comment by z3APA3A** [comment]
0.8 branch is not longer developed, only security-related bugs will be fixed for 0.8 branch. Next stable release will be 0.9.0 (current master). Transparent should work with IPv6 in both devel and master branches, but I have no chance to test it.
